editor/src/document/document_message_handler.rs

```diff
@@ -89,10 +113,14 @@ impl DocumentsMessageHandler {
 
 impl Default for DocumentsMessageHandler {
 	fn default() -> Self {
+		let mut map: HashMap<u64, DocumentMessageHandler> = HashMap::with_capacity(1);
```

Comment:
`documents` or `documents_map` would probably be a more descriptive name here, since "map" is just the name of the data structure, not what the data represents. Sorry to nitpick— on the flip side, these superficial comments indicate that the core work in this PR is without flaws, great job!